velox/docs/functions/spark/string.rst

```diff
@@ -179,6 +179,22 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
         SELECT substring('Spark SQL', -10, 3); -- "Sp"
         SELECT substring('Spark SQL', -20, 3); -- ""
 
+.. spark:function:: substring_index(string, delim, count) -> varchar
+
+    Returns the substring from ``string`` before ``count`` occurrences of the delimiter ``delim``.
```

Comment:
Would you clarify how this function handles overlapping 'delim'?

substring_index('aaaaa', 'aa', 2) ?

Also, let's clarify that 'string' can be VARCHAR or VARBINARY and return type matches type of 'string'.

I assume the implementation doesn't validate that string or delim is valid UTF-8. Just want to double check.

Reply:
Updated.